x-pack/test/api_integration/apis/guided_onboarding/get_state.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import {
+  testGuideStep1ActiveState,
+  testGuideNotActiveState,
+} from '@kbn/guided-onboarding-plugin/public/services/api.mocks';
+import { guidedSetupSavedObjectsType } from '@kbn/guided-onboarding-plugin/server/saved_objects/guided_setup';
+import type { GuideState, GuideId } from '@kbn/guided-onboarding';
+import type { FtrProviderContext } from '../../ftr_provider_context';
+
+const testGuideA = testGuideStep1ActiveState;
+const testGuideB = { ...testGuideNotActiveState, guideId: 'testGuideB' as GuideId }; // different guideId
+
+export default function testGetState({ getService }: FtrProviderContext) {
+  const supertest = getService('supertest');
+  const kibanaServer = getService('kibanaServer');
+
+  describe('GET /api/guided_onboarding/state', () => {
+    afterEach(async () => {
+      // Clean up saved objects
+      await kibanaServer.savedObjects.clean({ types: [guidedSetupSavedObjectsType] });
+    });
+
+    const createGuides = async (guides: GuideState[]) => {
+      for (const guide of guides) {
+        await kibanaServer.savedObjects.create({
+          type: guidedSetupSavedObjectsType,
+          id: guide.guideId,
+          overwrite: true,
+          attributes: guide,
+        });
+      }
+    };
+
+    it('should return the state for all guides', async () => {
+      // Create two guides to return
+      await createGuides([testGuideA, testGuideB]);
+
+      const response = await supertest.get('/api/guided_onboarding/state').expect(200);
+      expect(response.body.state.length).to.eql(2);
+      expect(response.body).to.eql({
+        state: [testGuideA, testGuideB],
+      });
+    });
+
+    it('should return the state for the active guide with query param `active=true`', async () => {
+      await createGuides([testGuideA, { ...testGuideB, isActive: false }]);
+
+      const response = await supertest
+        .get('/api/guided_onboarding/state')
+        .query({ active: true })
+        .expect(200);
+      expect(response.body).to.eql({ state: [testGuideA] });
+    });
+
+    it('should return state from saved object if it exists', async () => {
```

Comment:
I think this test case is covered by `should return the state for all guides` but let me know if I missed something 😅